compair/static/modules/assignment/assignment-module.js

```diff
@@ -673,21 +739,34 @@ module.factory( "AssignmentPermissions", function (){
                     'hasCompareDueDate' : assignment.compare_end,
 
                     // self-eval
-                    'canSelfEval'       : assignment.status.answers.answered && assignment.status.comparisons.left == 0,
-                    'needsSelfEval'     : assignment.status.comparisons.left == 0 && assignment.self_evaluation_needed, // && canCompare (added below)
+                    'isSelfEvalPeriod'  : assignment.self_eval_period,
+                    'canSelfEval'       : assignment.status.answers.answered, // more below
+                    'needsSelfEval'     : assignment.self_eval_period && assignment.self_evaluation_needed && !canManageAssignment && isAfterAnswer,
```

Comment:
Need to handle when there is no self-eval start/end (so comparisons for the user must be completed first)

Reply:
Your right, sorry I missed it